packages/cashier/src/pages/payment-agent/payment-agent-receipt/__tests__/payment-agent-receipt.spec.js

```diff
@@ -3,80 +3,105 @@ import PaymentAgentReceipt from '../payment-agent-receipt';
 import { fireEvent, render, screen } from '@testing-library/react';
 import { createBrowserHistory } from 'history';
 import { Router } from 'react-router';
-import { routes } from '@deriv/shared';
+import { isMobile, routes } from '@deriv/shared';
 
 jest.mock('Stores/connect.js', () => ({
     __esModule: true,
     default: 'mockedDefaultExport',
     connect: () => Component => Component,
 }));
 
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(() => false),
+}));
+
+jest.mock('Pages/payment-agent/payment-agent-disclaimer', () => () => <div>PaymentAgentDisclaimer</div>);
+
 describe('<PaymentAgentReceipt />', () => {
-    const mockProps = () => ({
+    const history = createBrowserHistory();
+    const props = {
         currency: 'USD',
+        is_from_derivgo: false,
         loginid: 'CR90000170',
         receipt: {
             amount_transferred: '20.00',
             payment_agent_email: 'reshma+cr1@binary.com',
             payment_agent_id: 'CR90000089',
             payment_agent_name: 'Ms QA script reshmacrcdD',
-            payment_agent_phone: '+62417522087',
-            payment_agent_url: 'https://deriv.com/',
+            payment_agent_phone: [{ phone_number: '+62417522087' }],
+            payment_agent_url: [{ url: 'https://deriv.com/' }],
         },
         resetPaymentAgent: jest.fn(),
-    });
+    };
 
     it('should show the proper text/messages', () => {
-        const history = createBrowserHistory();
-        const props = mockProps();
         render(
             <Router history={history}>
                 <PaymentAgentReceipt {...props} />
             </Router>
         );
-        expect(screen.getByText('Your funds have been transferred')).toBeInTheDocument();
-        expect(screen.getByText('20.00 USD')).toBeInTheDocument();
-        expect(screen.getByText('USD')).toBeInTheDocument();
-        expect(screen.getByText('CR90000170')).toBeInTheDocument();
-        expect(screen.getByText('Ms QA script reshmacrcdD')).toBeInTheDocument();
-        expect(screen.getByText('CR90000089')).toBeInTheDocument();
-        expect(screen.getByText('IMPORTANT NOTICE TO RECEIVE YOUR FUNDS')).toBeInTheDocument();
+
+        const [view_transactions_btn, make_a_new_withdrawal_btn] = screen.getAllByRole('button');
+
+        expect(screen.getByText('You’ve transferred 20.00 USD')).toBeInTheDocument();
```

Comment:
@george-usynin-binary I think we should reconsider the way we write tests, Instead of checking for the text we should only check for the element to be on the screen.
If the requirement changes and we change this text then the test will fail, So whenever we change something in the code we should also update the test and we are just maintaining two different places without getting many benefits, Maybe I'm wrong tho 🤔

Reply:
Agree with @farzin-fs here. We should be checking for the element instead of text imo as well.